service/history/replicationTaskProcessor.go

```diff
@@ -202,40 +205,38 @@ Loop:
 			}
 
 			scope.UpdateGauge(metrics.LastRetrievedMessageID, float64(p.lastRetrievedMessageID))
-			p.noTaskBackoffRetrier.Reset()
+			p.noTaskRetrier.Reset()
 		case <-p.done:
+			p.logger.Info("ReplicationTaskProcessor shutting down.")
 			return
 		}
 	}
 }
 
-func (p *ReplicationTaskProcessor) processTask(replicationTask *r.ReplicationTask) {
-	var err error
+func (p *ReplicationTaskProcessor) processTask(replicationTask *r.ReplicationTask) error {
+	err := backoff.Retry(func() error {
+		return p.processTaskOnce(replicationTask)
```

Comment:
i would suggest copy pasting logic from worker (or move those logic to common package and reference it here)
ref: https://github.com/uber/cadence/blob/master/service/worker/replicator/replicationTask.go

Reply:
I'll remove the code in worker after we deprecate kafka.